datastore/additional_data/sources/additional_data_recipient_location.py

```diff
@@ -0,0 +1,77 @@
+class AdditionalDataRecipientLocation(object):
+    """Adds recipient* additional data to grant data
+
+    These fields are used as the best available Region and District backward compatibility
+    with grantnav and are sourced from locationLookup.
+
+    https://github.com/ThreeSixtyGiving/grantnav/issues/867#issuecomment-1332505360
+
+    Fields updated by this source.
+       "recipientWardNameGeoCode",
+       "recipientWardName",
+       "recipientDistrictName",
+       "recipientDistrictGeoCode",
+       "recipientRegionName",
+       "recipientCountryName",
+       "recipientLocation", # string of all the above
+    """
+
+    def update_additional_data(self, grant, additional_data):
+        # If we have locationLookup from beneficiaryLocation or we don't have any
+        # recipientRegionName take whatever data we can from locationLookup instead
+        try:
+            additional_data["recipientRegionName"] = additional_data["locationLookup"][
```

Comment:
Is it worth adding `recipientRegionGeoCode` field here for consistency with the other fields?

Reply:
Yeah happy to do that, my aim is at some point to fully deprecate the recipient* fields as they're now just duplicate data that is there so that we can avoid refactoring the geo stuff in GrantNav.

Reply:
Maybe we should leave it then - sounds like something we could use when those fields are deprecated. 

The only other thing I thought of was checking whether there's anything hardcoded in GrantNav based on the region name - as the regions will be changing names. The only part I found in the code was [this test data](https://github.com/ThreeSixtyGiving/grantnav/blob/f735bee152c369fc84d6e07c064fd00a81c11e99/dataload/test_data/grantnav-20180903134856.json#L240)